### ISSUE TITLE
Merge email and remote sharing

### ISSUE THREAD

**Issue by MariusBluem** [nextcloud/server#5334]
To prevent showing an email-adress (or Federated Cloud ID) twice in the sharing with user dropdown, I'd propose to merge them into one single "admin@cloud.example.com"-entry ... Clicking on it will first check whether there is an Federation server (remote, prioritized), and if not - It will use email as fallback method (because it makes no sense to share with somebody who as Nextcloud himself via email). 

Would fix messages like:

<img width="889" alt="bildschirmfoto 2017-06-11 um 09 11 02" src="https://user-images.githubusercontent.com/19711361/27009120-40f10588-4e86-11e7-92ac-d8e4cf12111c.png">

Currently:

- admin@cloud.example.com (remote)
- admin@cloud.example.com (email)

What do you think? @schiessle @jancborchardt

**Comment by schiessle** [comment]
I don't think that this is a good idea. We can't verify if sending a mail was successful or not. This means that in case the remote server is down or you just have a typo in the federated cloud ID we will fall back to the email, the email will most likely fail and the user who created the share will think that everything was OK because we create a share entry in the share list.

**Comment by jancborchardt** [comment]
Why don't we always do both?

In the case of a federated share it's nice to also receive a mail confirmation (and fallback link to add it to your Nextcloud).

And in the case of a mail share they will be able to add it to their Nextcloud anyway via the share link - so we might as well send the share via federated sharing anyway.

**Comment by MariusBluem** [comment]
@jancborchardt & @MorrisJobke ... as discussed too :)

**Comment by skjnldsv** [comment]
@juliushaertl isn't it fixed with the new sharing ui?

**Comment by jancborchardt** [comment]
@rullzer @MorrisJobke this is something we want, correct? It is mostly a backend question.

**Comment by MorrisJobke** [comment]
> @rullzer @MorrisJobke this is something we want, correct? It is mostly a backend question.

I guess it needs a bit more thinking, but yes - mostly backend stuff.

**Comment by szaimen** [comment]
I am going to close this since there doesn't seem to be a lot of interest (no upvotes) and no respond since around 2 years. Please reopen if you still want to implement this.
